OpenRA.Mods.Common/UtilityCommands/UpgradeRules.cs

```diff
@@ -1674,6 +1674,41 @@ internal static void UpgradeActorRules(ModData modData, int engineVersion, ref L
 					}
 				}
 
+				if (engineVersion < 20180309)
+				{
+					if (node.Key == "ParaDrop")
+					{
+						var soundNodePD = node.Value.Nodes.FirstOrDefault(n => n.Key == "ChuteSound");
```

Comment:
Code style question - Is there a reason there's not a helper method to deal with repetition?

It seems like the following pattern could express your intent for all these cases...
AddOnMissingNode(nodes, key, value), used like:
AddOnMissingNode(node.Value.Nodes, "ChuteSound","chute1.aud")

Reply:
> Code style question - Is there a reason there's not a helper method to deal with repetition of conditionally adding?

Apart from that nobody has bothered writing one yet, not really.

@pchote has started writing a (currently about half-finished) complete replacement for the current upgrade rule system though, so at this point it's questionable whether such a helper method would have any tangible benefit for the (hopefully short) remaining lifetime of the current system.

Reply:
For reference, this is how it looks on the new system: https://github.com/pchote/OpenRA/blob/4f779e91c2712709776dbb9cd22662defb59f5a0/OpenRA.Mods.Common/UpdateRules/Rules/DefineSoundDefaults.cs

However, note that this works slightly differently: it prints the locations that *might* want to be changed, rather than changing everything and then relying on the person running the command implicitly knowing that they need to undo most of them.